control/cephutils.py

```diff
@@ -21,6 +22,41 @@ def __init__(self, config):
         self.logger = GatewayLogger(config).logger
         self.ceph_conf = config.get_with_default("ceph", "config_file", "/etc/ceph/ceph.conf")
         self.rados_id = config.get_with_default("ceph", "id", "")
+        self.integer_list = []
+        self.last_sent = time.time()
+
+    def get_number_created_gateways(self, pool, group):
+        now = time.time()
+        if (now - self.last_sent) < 10 and self.integer_list :
+             self.logger.info(f" Caching responce of the monitor: {self.integer_list} ")
+             return self.integer_list
+        else :
+            try:
+                self.integer_list = []
+                self.last_sent = now
+                with rados.Rados(conffile=self.ceph_conf, rados_id=self.rados_id) as cluster:
```

Comment:
Instead of duplicating this code we could add a function:
```
def execute_ceph_monitor_command(self, cmd):
    with rados.Rados(conffile=self.ceph_conf, rados_id=self.rados_id) as cluster:
        rply = cluster.mon_command(cmd, b'')
        return rply
```

And then use:
```
    str = '{' + f'"prefix":"nvme-gw show", "pool":"{pool}", "group":"{group}"' + '}'
    self.logger.debug(f"nvme-show string: {str} ")
    rply = self.execute_ceph_monitor_command(str)
    .
    .
    .
```

And something similar in fetch_and_display_ceph_version()

Reply:
done